src/librustc_typeck/check/mod.rs

```diff
@@ -4912,7 +4912,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 }
                 Some(Node::Ctor(hir::VariantData::Tuple(fields, _))) => {
                     sugg_call = fields.iter().map(|_| "_").collect::<Vec<_>>().join(", ");
-                    match hir.as_local_hir_id(def_id).and_then(|hir_id| hir.def_kind(hir_id)) {
+                    match def_id.try_to_local().and_then(|hir_id| hir.def_kind(hir_id)) {
```

Comment:
This is weird, it should just use `tcx.def_kind(def_id)`.

Reply:
Ah, I was wondering about that. It seems like `self.tcx.def_kind(...)` ought to work.

Reply:
Hmm... actually, as currently written, it is resilient against non-local `DefId`s (because `try_to_local` would be `None`). Is that important in this case?

Reply:
It seems to work ok just using `tcx.def_kind`